<div class="container">

<table style="width: 100%;"><tr>
<td>Transcriptomics_Rosettes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transcriptomics Rosettes</h2>

<h3>Description</h3>

<p>A dataset containing all the transcripts obtained by RNA-seq performed, according to the standard Illumina protocols,
on rosettes of five <em>A. thaliana</em> genotypes at two growth temperatures.
See Ecotype and Temperature for more information.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("Transcriptomics_Rosettes")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 19763 variables.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Load the dataset
data("Transcriptomics_Rosettes")

# Look at data frame dimensions
dim(Transcriptomics_Rosettes)

# Look at the first rows and columns
head(Transcriptomics_Rosettes[,c(1:10)])

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# PCA on transcriptomics
res.pca &lt;- prcomp(Transcriptomics_Rosettes, center = TRUE, scale. = TRUE)
plot(res.pca$x[,"PC1"], res.pca$x[,"PC2"], pch = 19, xlab = "PC1", ylab = "PC2", lwd = 5,
     main = "Individuals' plot (1 x 2) - PCA on Rosettes' Transcriptomics",
     col = colors)
text(res.pca$x[,"PC1"], res.pca$x[,"PC2"], labels = row.names(res.pca$x), cex = 0.8, pos = 3)

</code></pre>


</div>