<div class="container">

<table style="width: 100%;"><tr>
<td>linkMap.wgaim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot a genetic linkage map with QTL</h2>

<h3>Description</h3>

<p>Neatly plots the genetic linkage map with marker locations,
marker names and highlights QTL with their associated flanking markers
obtained from a <code>wgaim</code> model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'wgaim'
linkMap(object, intervalObj, chr, chr.dist,
    marker.names = "markers", flanking = TRUE, list.col = list(q.col = "light blue",
    m.col = "red", t.col = "light blue"), list.cex = list(t.cex = 0.6,
    m.cex = 0.6), trait.labels = NULL, tick = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>object of class <code>"wgaim"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intervalObj</code></td>
<td>
<p>object of class <code>"cross"</code> or <code>"interval"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>character string naming the subset of chromosomes to plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.dist</code></td>
<td>
<p>a list containing named elements <code>"start"</code> and
<code>"end"</code> containing the start and end distances in cM the
genetic map should be subsetted by. Each of these may also be
a vector of distances equal to the length of the number of linkage
groups to be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker.names</code></td>
<td>
<p>a character string naming the type of marker
information to plot. If <code>"dist"</code> then distances names
plotted alongside each chromosome on the left. If <code>"markers"</code>
then marker names are plotted instead. Defaults to
<code>"markers"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flanking</code></td>
<td>
<p>logical value. If <code>TRUE</code> then only plot marker
names or distances for flanking markers of the QTL. Defualts to <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>list.col</code></td>
<td>
<p>named list of colours used to highlight the QTL regions and
their flanking markers. <code>q.col</code> is the color of the QTL
regions. <code>m.col</code> is the color the flanking markers. <code>t.col</code>
is the color of the trait name used in the model object (see
<code>par</code> for colour options)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>list.cex</code></td>
<td>
<p>a named list object containing the character expansion
factors for the marker names <code>m.cex</code> and the trait labels <code>t.cex</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait.labels</code></td>
<td>
<p>character string naming the trait used in the
model object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tick</code></td>
<td>
<p>logical value. If <code>TRUE</code> then an axis with tick marks are
generated for the chromosome names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments passed to the <code>plot()</code> function to set up the plot
region and plot any symbols if required</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This plotting procedure builds on <code>linkMap.cross()</code> by adding the
QTL regions to the map and highlighting the appropriate markers obtained
from a fit to <code>wgaim</code>. If the linkage map is subsetted and QTL
regions fall outside the remaining map a warning will be given that
the QTL have been omitted from the display.
</p>
<p>The <code>list.col</code> arguments <code>q.col</code>, <code>m.col</code> and
<code>t.col</code> have been added for personal colour highlighting of the QTL
regions, flanking markers and trait names. For greater flexibility the
procedure may also be given the usual <code>col</code> argument that will be
passed to the other markers.
</p>
<p>The <code>list.cex</code> argument can be used to manipulate the character expansion of
the marker names using <code>m.cex</code> or the character expansion of the
<code>trait.labels</code> using <code>t.cex</code>. If a <code>"marker"</code> analysis has been
performed then <code>pch</code> is used to plot a symbol at the
location of the QTL. This character can be changed using the usual
arguments such as <code>pch</code> or <code>cex</code> that are passed through the
... argument.
</p>


<h3>Value</h3>

<p>For an <code>"interval"</code> analysis, the genetic linkage map is
plotted with shaded QTL regions and highlighted flanking markers. For
a <code>"marker"</code> analysis, a symbol is placed at the QTL locations and
the markers are highlighted.
</p>


<h3>Author(s)</h3>

<p>Julian Taylor</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical and
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code>linkMap.cross</code>, <code>wgaim</code></p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 
# fit wgaim model

yield.qtl &lt;- wgaim(yield.fm, intervalObj = genoRxK, merge.by = "Genotype",
                  trace = "trace.txt", na.action = na.method(x = "include"))

# plot QTL

linkMap(yield.qtl, genoRxK, list.col = list(m.col = "red"), col = "gray")


## End(Not run)
</code></pre>


</div>