<div class="container">

<table style="width: 100%;"><tr>
<td>cross2int</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert a cross genetic object to an interval object</h2>

<h3>Description</h3>

<p>Converts an object of class "<code>cross</code>" to an object with class
"<code>interval</code>". The function also imputes missing markers.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cross2int(object, impute = "MartinezCurnow", consensus.mark = TRUE,
     id = "id", subset = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>an object of class <code>"cross"</code> that inherits one of the class
structures <code>"bc"</code>, <code>"dh"</code>, <code>"f2"</code>, <code>"riself"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute</code></td>
<td>
<p>a character string determining how missing values in
the linkage map should be imputed. If <code>"Broman"</code>, then missing values are
imputed according to Bromans rules. If <code>"MartinezCurnow"</code> then
missing values are imputed according to the rules of Martinez &amp;
Curnow (1994) (see reference list). The default is <code>"MartinezCurnow"</code> (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>consensus.mark</code></td>
<td>
<p>logical value. If <code>TRUE</code> co-locating marker sets
are condensed to form consensus markers (see Details). Defaults to
<code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>a character string or name of the unique identifier for each row of genotype
data (see Details). Defaults to <code>"id"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset</code></td>
<td>
<p>a possible character vector naming the subset of
chromosomes to be returned. Defaults to <code>NULL</code> implying return
all chromosomes.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function provides the conversion of genetic data objects that have
already been generated using <code>read.cross()</code> from Bromans <span class="pkg">qtl</span>
package, to <code>"interval"</code> objects ready for use with
<code>wgaim</code>. Users should be aware that this function is restricted to
certain populations.  <code>object</code> must inherit one of the class
structures <code>"bc"</code>, <code>"dh"</code>, <code>"f2"</code>, <code>"riself"</code>.
</p>
<p>During the conversion process three important linkage map attributes are
assessed.
</p>

<ol>
<li>
<p> The map may be subsetted using the <code>subset</code> argument
</p>
</li>
<li>
<p> If <code>consensus.mark = TRUE</code> then co-located marker sets are reduced
to form single consensus markers before missing values are
imputed. The marker similarity is determined by
the genetic distances that are given in the map component for each linkage
group. If a set of markers co-locate the name of the first marker is
chosen and a single consensus marker is determined by coalescing the
genetic information from all markers in the set. A "(C)" is placed
after the marker name for easy identification. The markers removed
from each set are returned with the object and placed under
<code>"colocated.markers"</code> for inspection if required.
</p>
</li>
<li>
<p> Missing values are imputed according to
the argument given by <code>impute</code>. This imputation results in a
complete version of the marker data for each chromosome which is then
used to create the interval data component "<code>interval.data</code>". The complete
marker data for each chromosome can be obtained from the "<code>imputed.data</code>" element of the
returned list. It is therefore also possible to perform whole genome <em>marker</em>
analysis using <code>wgaim</code>. See <code>wgaim.asreml</code> for more details.
</p>
</li>
</ol>
<h3>Value</h3>

<p>a list of class <code>"cross"</code> that also inherits the class
<code>"interval"</code>. The list contains the following components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>

<p>A list with elements named by the corresponding names of the
chromosomes. Each chromosome is itself a list with six
elements: <code>"data"</code> is the actual estimated map matrix with rows
as individuals named by <code>"id"</code> and markers as columns;
<code>"map"</code> is a vector of marker positions on the
corresponding chromosome; <code>"imputed.data"</code> is identical to <code>"data"</code>
matrix but with all NAs replaced by imputed values according to the
rules of <code>"impute"</code>; <code>"dist"</code> contains the genetic
distance between adjacent markers or the genetic distances of the
intervals; <code>"theta"</code> contains the recombination fractions for
each interval; <code>"interval.data"</code> contains the recalculated intervals
based on the recombination fractions and the missing marker information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colocated.markers</code></td>
<td>

<p>If <code>consensus.mark = TRUE</code>, a four column data frame containing
stacked binned sets of co-located markers. In each binned set the first
marker is the unique consensus marker name used in the linkage map and the
others are the co-located marker names that were omitted. Additionally
for each binned set, the data frame also contains linkage group
identification and marker position information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>

<p>A data.frame of phenotypic information with rows as individuals read
in from <code>read.cross</code>. A copy of the column named by the <code>"id"</code> argument
can be found here (see the help for the <code>read.cross() function</code>).</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Julian Taylor and Ari Verblya</p>


<h3>References</h3>

<p>Martinez, O., Curnow. R. N. (1994) Missing markers when estimating
quantitative trait loci using regression mapping. <em>Heredity</em>,
<b>73</b>, 198-206.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical And
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code>read.cross</code></p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 
# read in linkage map from a rotated .CSV file with "id" as the
# identifier for each unique row

wgpath &lt;- system.file("extdata", package = "wgaim")
genoSxT &lt;- read.cross("csvr", file="genoSxT.csv", genotypes=c("AA","BB"),
         na.strings = c("-", "NA"), dir = wgpath)
genoSxT &lt;- cross2int(genoSxT, impute="MartinezCurnow", id = "id")

# plot linkage map

linkMap(genoSxT, cex = 0.5)


## End(Not run)
</code></pre>


</div>