<div class="container">

<table style="width: 100%;"><tr>
<td>WGScan.Region</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Scan the association between an quantitative/dichotomous outcome variable and a region by score type statistics allowing for multiple functional annotation scores.
</h2>

<h3>Description</h3>

<p>Once the preliminary work is done by "WGScan.prelim()", this function scan a target region. This function is often used for candidate region analyses.
</p>


<h3>Usage</h3>

<pre><code class="language-R">WGScan.Region(result.prelim,G,pos,Gsub.id=NULL,Z=NULL,MAF.weights='beta',
test='combined',window.size=c(5000,10000,15000,20000,25000,50000),MAF.threshold=1,
impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>result.prelim</code></td>
<td>

<p>The output of function "WGScan.prelim()"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>

<p>Genetic variants in the target region, an n*p matrix where n is the subject ID and p is the total number of genetic variants.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>

<p>The positions of genetic variants, an p dimensional vector. Each position corresponds to a column in the genotype matrix.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. Each ID corresponds to a row in the genotype matrix. This is used to match phenotype with genotype. The default is NULL, where the matched phenotype and genotype matrices are assumed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>

<p>Weight matrix for functional annotations, an p*q matrix where p is the total number of genetic variables and q is the number of weights. This is used to incorperate functional annotations. The default is NULL, where minor allele frequency weighted (see MAF.weights) dispersion and/or burden tests are applied.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF.weights</code></td>
<td>

<p>Minor allele frequency based weight. Can be 'beta' to up-weight rare variants or 'equal' for a flat weight. The default is 'beta'.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>

<p>Can be 'dispersion', 'burden' or 'combined'. The test is 'combined', both dispersion and burden tests are applied. The default is 'combined'.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window.size</code></td>
<td>

<p>Candidate window sizes in base pairs. The default is c(5000,10000,15000,20000,25000,50000). Note that extemely small window size (e.g. 1) requires large sample size.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF.threshold</code></td>
<td>

<p>Threshold for minor allele frequency. Variants above MAF.threshold are ignored. The default is 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be "random", "fixed" or "bestguess". Given the estimated allele frequency, "random" simulates the genotype from binomial distribution; "fixed" uses the genotype expectation; "bestguess" uses the genotype with highest probability.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.marker</code></td>
<td>

<p>Number of tested variants in the window (heterozygous variants below MAF threshold).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window.summary</code></td>
<td>

<p>Results for all windows. Each row presents a window, including chromosome number, start position, end position,dispersion p-value(s), burden p-values(s).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>

<p>Estimated number of effective tests.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>

<p>Estimated threshold, 0.05/M. This threshold is for windows tested in this particular region.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>

<p>P-value of entire region.
</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## WGScan.prelim does the preliminary data management.
# Input: Y, X (covariates)
## WGScan.Region scans a region.
# Input: G (genetic variants), pos (position) Z (weights) and result of WGScan.prelim

library(WGScan)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# pos: positions of genetic variants, p dimention vector
# Z: functional annotation matrix, p by q matrix

data(WGScan.example)
Y&lt;-WGScan.example$Y;X&lt;-WGScan.example$X
G&lt;-WGScan.example$G;pos&lt;-WGScan.example$pos
Z&lt;-WGScan.example$Z

# Preliminary data management
result.prelim&lt;-WGScan.prelim(Y,X=X,out_type="C",B=5000)

# Scan the region with functional annotations defined in Z
result&lt;-WGScan.Region(result.prelim,G,pos,Z=Z)
</code></pre>


</div>