<div class="container">

<table style="width: 100%;"><tr>
<td>nutrimouse</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Nutrimouse Data</h2>

<h3>Description</h3>

<p>The <code>nutrimouse</code> dataset is a collection of gene expression and lipid measurements 
collected in a nutrigenomic study in the mouse studying 40 animals by Martin et al. (2007).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(nutrimouse)
</code></pre>


<h3>Format</h3>

<p>A list containing the following components:
</p>
<p><code>gene</code>  collects gene expression of 120 genes in liver tissue for 40 mice.
</p>
<p><code>lipid</code> collects concentrations of 21 lipids for 40 mice.
</p>
<p><code>diet</code> describes the diet of each mouse ("coc", "fish", "lin", "ref", or  "sun").
</p>
<p><code>genotype</code> describes the genotype of each mouse: wild type ("wt") or PPARalpha deficient ("ppar").
</p>


<h3>Details</h3>

<p>This data set is used to illustrate CCA-based data integration
in Jendoubi and Strimmer (2019) and is also described in Gonzalez et al. (2008).
</p>


<h3>Source</h3>

<p>The original data are available in the <code>CCA</code> package by Gonzalez et al. (2008), see their function <code>nutrimouse</code>.
</p>


<h3>References</h3>

<p>Gonzalez, I., Dejean, S., Martin, P.G.P,  Baccini, A.
CCA: an R package to extend canonical correlation analysis. J. Statist. Software 23:1–13 (2008)
</p>
<p>Jendoubi, T., Strimmer, K.: A whitening approach to probabilistic canonical correlation analysis for omics data integration. BMC Bioinformatics 20:15 
&lt;DOI:10.1186/s12859-018-2572-9&gt;
</p>
<p>Martin, P.G.P., Guillou, H., Lasserre, F., Dejean, S., Lan, A., Pascussi, J.-M., Cristobal, M.S., Legrand, P., Besse, P., Pineau, T.: Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed
through a multigenomic study. Hepatology 54, 767–777 (2007)
&lt;DOI:10.1002/hep.21510&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load whitening library
library("whitening")

# load nutrimouse data set
data(nutrimouse)

dim(nutrimouse$gene) # 40 120
dim(nutrimouse$lipid) # 40 21
levels( nutrimouse$diet ) #  "coc"  "fish" "lin"  "ref"  "sun"
levels( nutrimouse$genotype ) # "wt"   "ppar"
</code></pre>


</div>