<div class="container">

<table style="width: 100%;"><tr>
<td>estimate_m_rd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimate heterozygote miscall rate for different read depth categories (no nulls)</h2>

<h3>Description</h3>

<p>To see how this Rcpp function is applied, see the code in
<code>infer_m</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">estimate_m_rd(Y, R, init_m, num_cats, p_prior, m_prior, num_reps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>the 012,-1 matrix that is N x L giving the observed genotypes of the N individuals
at L SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>R</code></td>
<td>
<p>integer matrix that is N x L giving the read depth categories.  These must be indexed from
1 up to num_cats.  Missing data should be -1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init_m</code></td>
<td>
<p>starting value for m.  Typically you might want to use the m estimated
from init_m</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_cats</code></td>
<td>
<p>the number of read depth categories.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_prior</code></td>
<td>
<p>two-vector that holds the beta parameters for a prior on the
allele frequency for each locus.  Typical value is c(0.5, 0.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m_prior</code></td>
<td>
<p>two-vector that holds the beta parameters for a prior on the
heterozygote miscall rate for each locus.  Typical value is c(0.5, 0.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_reps</code></td>
<td>
<p>the number of MCMC sweeps to do.</p>
</td>
</tr>
</table>
</div>