<div class="container">

<table style="width: 100%;"><tr>
<td>whoa</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>whoa: Evaluation of genotyping error in genotype by sequencing data</h2>

<h3>Description</h3>

<p>The name is found in the capitals here:  <code>W</code>here's my <code>H</code>eterozygotes?! <code>O</code>bservations
of genotyping <code>A</code>ccuracy.
</p>


<h3>the <code>whoa</code> main high-level functions</h3>

<p>The main function in the package whoa is <code>infer_m</code>.
This function infers the heterozygote miscall rate (the rate at
which true heterozygotes have been miscalled as homozygotes in
genotype-by-sequencing data) for calls made upon genotypes falling
within different read depth categories.
</p>
<p>The output from  <code>infer_m</code> is easily plotted by passing the
m_posteriors component of the output list from <code>infer_m</code> into
<code>posteriors_plot</code>.
</p>


<h3>example data</h3>

<p>The package comes with a small data set, <code>lobster_buz_2000</code>, which
was read in from a variant call format (VCF) file and is now stored in the package as a ‘vcfR’ object.
</p>


</div>