<div class="container">

<table style="width: 100%;"><tr>
<td>geno_freqs_scatter</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>return a ‘ggplot2’ plot object of observed and expected genotype freqs</h2>

<h3>Description</h3>

<p>return a ‘ggplot2’ plot object of observed and expected genotype freqs
</p>


<h3>Usage</h3>

<pre><code class="language-R">geno_freqs_scatter(gfc, alpha = 0.2, max_plot_loci = 500)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gfc</code></td>
<td>
<p>a tibble like that created by exp_and_obs_geno_freqs()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>the transparency (alpha) parameter to apply to the points
in the scatterplot. Default is 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_plot_loci</code></td>
<td>
<p>By default this plots only 500 loci, sampled
randomly, to keep ‘ggplot2’ taking forever to plot, for example, 100K
points.  If you want to plot all the points, set this to a number
larger than the number of single nucleotide polymorphisms (SNPs) in the data set.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R"># get the expected and observed geno freqs
gfreqs &lt;- exp_and_obs_geno_freqs(lobster_buz_2000)
g &lt;- geno_freqs_scatter(gfreqs)

# now g is a 'ggplot2' object.
</code></pre>


</div>